libraries/AP_OpenDroneID/AP_OpenDroneID.h

```diff
@@ -0,0 +1,360 @@
+ *  BlueMark Innovations BV, Roel Schiphorst
+ *  Contributors: Tom Pittenger, Josh Henderson
+ *  Parts of this code are based on/copied from the Open Drone ID project https://github.com/opendroneid/opendroneid-core-c
+ *
+ * The code has been tested with the BlueMark DroneBeacon MAVLink transponder running this command in the ArduPlane folder:
+ * sim_vehicle.py --wipe-eeprom --console --map -A --serial1=uart:/dev/ttyUSB1:9600
+ * (and a DroneBeacon MAVLink transponder connected to ttyUSB1)
+ *
+ * The Remote ID implementation expects a transponder that caches the received MAVLink messages from ArduPilot
+ * and transmits them at the required intervals. So static messages are only send once to the transponder.
+ */
+
+#pragma once
+
+#ifndef AP_OPENDRONEID_ENABLED
+#define AP_OPENDRONEID_ENABLED 1
+#endif
+
+#include <AP_Math/AP_Math.h>
+#include <AP_Param/AP_Param.h>
+#include <GCS_MAVLink/GCS_MAVLink.h>
+#include <AP_Common/Location.h>
+
+#define ODID_ID_SIZE 20
+#define ODID_STR_SIZE 23
+
+#define ODID_MIN_DIR         0       // Minimum direction
+#define ODID_MAX_DIR         360     // Maximum direction
+#define ODID_INV_DIR         361     // Invalid direction
+#define ODID_MIN_SPEED_H     0       // Minimum speed horizontal
+#define ODID_MAX_SPEED_H     254.25f // Maximum speed horizontal
+#define ODID_INV_SPEED_H     255     // Invalid speed horizontal
+#define ODID_MIN_SPEED_V     (-62)   // Minimum speed vertical
+#define ODID_MAX_SPEED_V     62      // Maximum speed vertical
+#define ODID_INV_SPEED_V     63      // Invalid speed vertical
+#define ODID_MIN_ALT         (-1000) // Minimum altitude
+#define ODID_MAX_ALT         31767.5f// Maximum altitude
+#define ODID_INV_ALT         ODID_MIN_ALT // Invalid altitude
+#define ODID_MAX_TIMESTAMP   (60 * 60)
+#define ODID_INV_TIMESTAMP   0xFFFF  // Invalid, No Value or Unknown Timestamp
+#define ODID_MAX_AREA_RADIUS 2550
+#define ODID_AREA_COUNT_MIN  1
+#define ODID_AREA_COUNT_MAX  65000
+
+class AP_OpenDroneID {
+public:
+    AP_OpenDroneID();
+
+    /* Do not allow copies */
+    CLASS_NO_COPY(AP_OpenDroneID);
+
+    // parameter block
+    static const struct AP_Param::GroupInfo var_info[];
+
+    bool transmit_basic_ID_message;  //there are small differences in Remote ID standards https://github.com/opendroneid/opendroneid-core-c#comparison
+    bool transmit_location_message;  //this means that for instance in the EU the operator ID message is compulsory, other areas it is optional
+    bool transmit_self_ID_message;   //also the basic ID message (serial number) could be handled by the transponder itself
+    bool transmit_system_message;    //so we need to have bool variables to control which MAVLink packages are sent
+    bool transmit_operator_id_message;
+
+    void init();
+    void update();
+    void send_dynamic_out();
+    void send_static_out();
+
+    // get methods (not sure if we need those)
+    MAV_ODID_ID_TYPE get_id_type() {
+        return _id_type;
+    }
```

Comment:
You are correct we will mostly likely delete these all but until we are sure I've fixed the style.

I haven't gotten into the details of where the fields will come and go yet.

And made them `const`

```suggestion
    // get methods (not sure if we need those)
    MAV_ODID_ID_TYPE get_id_type() const { return _id_type; }
```